Build failure: graalvm-ce 22.0.0 fails to build on x86_64-darwin

### Steps To Reproduce

Steps to reproduce the behavior:
1. build graalvm-ce

### Build log

```
[8/8] Creating image...       [**]                                                                       (0.0s @ 0.31GB)
------------------------------------------------------------------------------------------------------------------------
                        2.5s (6.6% of total time) in 245 GCs | Peak RSS: 0.77GB | CPU load: 7.20
------------------------------------------------------------------------------------------------------------------------
Produced artifacts:
 /private/tmp/nix-build-graalvm-ce-22.0.0.drv-0/svm_err_b_20240326T121843.687_pid12649.md (build_info)
========================================================================================================================
Failed generating 'helloworld' after 36.1s.

The build process encountered an unexpected error:

java.lang.RuntimeException: There was an error linking the native image: Linker command exited with 1

Linker command executed:
/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wl,-U,___darwin_check_fd_set_overflow -Wl,-no_compact_unwind -Wl,-exported_symbols_list -Wl,/private/tmp/nix-build-graalvm-ce-22.0.0.drv-0/
SVM-6603972289861419722/exported_symbols.list -Wl,-x -arch x86_64 -o /private/tmp/nix-build-graalvm-ce-22.0.0.drv-0/helloworld helloworld.o /nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/svm/cl
ibraries/darwin-amd64/liblibchelper.a /nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libnet.a /nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/svm/clibrarie
s/darwin-amd64/libdarwin.a /nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libnio.a /nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libj
ava.a /nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libzip.a /nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/svm/clibraries/darwin-amd64/libjvm.a -v -L/pr
ivate/tmp/nix-build-graalvm-ce-22.0.0.drv-0/SVM-6603972289861419722 -L/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64 -L/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce
-22.0.0/lib/svm/clibraries/darwin-amd64 -ldl -lpthread -Wl,-framework,Foundation -lz

Linker command output:
clang version 16.0.6
Target: x86_64-apple-darwin
Thread model: posix
InstalledDir: /nix/store/qii4shz46sj38q2wdawiagshj4n1f0lr-clang-16.0.6/bin
 "/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/ld" -demangle -lto_library /nix/store/qii4shz46sj38q2wdawiagshj4n1f0lr-clang-16.0.6/lib/libLTO.dylib -dynamic -arch x86_64 -platform_version
macos 10.12.0 10.12.0 -o /private/tmp/nix-build-graalvm-ce-22.0.0.drv-0/helloworld -L/private/tmp/nix-build-graalvm-ce-22.0.0.drv-0/SVM-6603972289861419722 -L/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce
-22.0.0/lib/static/darwin-amd64 -L/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/svm/clibraries/darwin-amd64 -L/nix/store/h1fj93a17dlpjm4dm9bxg46kjxhwmbgn-libcxx-16.0.6/lib -L/nix/store/7naprg1
cjba3xq7a7kdfs3m45nwhxrrs-compiler-rt-libc-16.0.6/lib -L/nix/store/mshw25dqlc3csddksf9q23d6j018gwys-zlib-1.3.1/lib -L/nix/store/gyk6bgdhz4z9xj285y0sa5sv2s96649k-objc4-709.1/lib -L/nix/store/h1fj93a17dlpjm4dm9bxg4
6kjxhwmbgn-libcxx-16.0.6/lib -L/nix/store/7naprg1cjba3xq7a7kdfs3m45nwhxrrs-compiler-rt-libc-16.0.6/lib -L/nix/store/mshw25dqlc3csddksf9q23d6j018gwys-zlib-1.3.1/lib -L/nix/store/gyk6bgdhz4z9xj285y0sa5sv2s96649k-ob
jc4-709.1/lib -L/nix/store/8vzfq1fzj4qim2imivxfyhnlbzx9j4zq-Libsystem-1238.60.2/lib -L/nix/store/cy8py0rq64fikvhaj1959svksl33dw9v-clang-16.0.6-lib/lib -L/nix/store/h1fj93a17dlpjm4dm9bxg46kjxhwmbgn-libcxx-16.0.6/l
ib -U ___darwin_check_fd_set_overflow -no_compact_unwind -exported_symbols_list /private/tmp/nix-build-graalvm-ce-22.0.0.drv-0/SVM-6603972289861419722/exported_symbols.list -x helloworld.o /nix/store/ilvxwjpcjmh8
3kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/svm/clibraries/darwin-amd64/liblibchelper.a /nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libnet.a /nix/store/ilvxwjpcjmh83kkif1x
c3ya1a0qjkx46-graalvm-ce-22.0.0/lib/svm/clibraries/darwin-amd64/libdarwin.a /nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libnio.a /nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx
46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libjava.a /nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libzip.a /nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/
lib/svm/clibraries/darwin-amd64/libjvm.a -ldl -lpthread -framework Foundation -lz /nix/store/jvy4anwliaqn66yf8phdc2ai13977mwa-apple-framework-CoreFoundation/Library/Frameworks/CoreFoundation.framework/CoreFoundat
ion.tbd -arch x86_64 -lSystem /nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/resource-root/lib/darwin/libclang_rt.osx.a -F/nix/store/jvy4anwliaqn66yf8phdc2ai13977mwa-apple-framework-CoreFoundati
on/Library/Frameworks -F/nix/store/f1aa4ab5ywndh9x0lx1afhjd8x5smfls-apple-framework-Foundation/Library/Frameworks -F/nix/store/zvhh5snnl073w6d58rh617kp41m3wp9v-apple-framework-ApplicationServices/Library/Framewor
ks -F/nix/store/2xsjlmhv7im39vzifa15z3skmw2d3f81-apple-framework-CoreGraphics/Library/Frameworks -F/nix/store/38kx73nm83m5xd5llqvwgmnf0i2z60zr-apple-framework-Accelerate/Library/Frameworks -F/nix/store/mji904fmdj
vzf09j0xdiw4ps80y60ndf-apple-framework-CoreWLAN/Library/Frameworks -F/nix/store/9ysb6j9i85v7vdls2jcvg7fly3r39vz4-apple-framework-SecurityFoundation/Library/Frameworks -F/nix/store/q9yrxxnkn9ry0nvf48p5fih6wg34wcs3
-apple-framework-IOBluetooth/Library/Frameworks -F/nix/store/qfz0c82zhf12x43jxcwhvfrg6f5d4sg1-apple-framework-CoreBluetooth/Library/Frameworks -F/nix/store/x54lc2llkvp2ak7s3kh0dq3yvlnid8as-apple-framework-IOKit/L
ibrary/Frameworks -F/nix/store/gs1c9qkz6b8al4aif7z8wgbl7hxvq1z0-apple-framework-IOSurface/Library/Frameworks -F/nix/store/5ih2g3461xn7n4dwjzfj4z9kp1va13gi-apple-framework-SystemConfiguration/Library/Frameworks -F
/nix/store/9l7ckyy8lrlgbcf7rf93iiikgxv6g3n9-apple-framework-Security/Library/Frameworks -F/nix/store/qhsgpzs9xsjrrmr2c2gbpgxl9lihkbnw-apple-framework-CoreServices/Library/Frameworks -F/nix/store/j9m3c20vw2sc3z706
r34s94whxirwr76-apple-framework-CFNetwork/Library/Frameworks -F/nix/store/hc95skcs2rqf6jyw01j075pljg4i060j-apple-framework-CoreAudio/Library/Frameworks -F/nix/store/l66hcpw1cpwrbks2gln8fijq7glmb15z-apple-framewor
k-CoreData/Library/Frameworks -F/nix/store/jvy4anwliaqn66yf8phdc2ai13977mwa-apple-framework-CoreFoundation/Library/Frameworks -F/nix/store/ii38vbr1rkkmyi0bpfma28n6kawm509z-apple-framework-DiskArbitration/Library/
Frameworks -F/nix/store/p05ad4h6lpkm3g7gqfg11zjkr2b13grm-apple-framework-NetFS/Library/Frameworks -F/nix/store/zfmqzjnykhdxg5s2ryalixc2avj3l6k7-apple-framework-OpenDirectory/Library/Frameworks -F/nix/store/3cikf6
yzkrfj1aar45s9q7mna9008y3i-apple-framework-ServiceManagement/Library/Frameworks -F/nix/store/jplkm2kw2ny8wayp7n7jd3hc3dqyi9xz-apple-framework-CoreText/Library/Frameworks -F/nix/store/b69ikp6xf9ljng8lvylia91rfpcwb
22d-apple-framework-ImageIO/Library/Frameworks -F/nix/store/qhhfwlbwaz5ac0xqsvvg0fn61h6rkwql-apple-framework-CoreFoundation/Library/Frameworks -F/nix/store/f1aa4ab5ywndh9x0lx1afhjd8x5smfls-apple-framework-Foundat
ion/Library/Frameworks -F/nix/store/zvhh5snnl073w6d58rh617kp41m3wp9v-apple-framework-ApplicationServices/Library/Frameworks -F/nix/store/2xsjlmhv7im39vzifa15z3skmw2d3f81-apple-framework-CoreGraphics/Library/Frame
works -F/nix/store/38kx73nm83m5xd5llqvwgmnf0i2z60zr-apple-framework-Accelerate/Library/Frameworks -F/nix/store/mji904fmdjvzf09j0xdiw4ps80y60ndf-apple-framework-CoreWLAN/Library/Frameworks -F/nix/store/9ysb6j9i85v
7vdls2jcvg7fly3r39vz4-apple-framework-SecurityFoundation/Library/Frameworks -F/nix/store/q9yrxxnkn9ry0nvf48p5fih6wg34wcs3-apple-framework-IOBluetooth/Library/Frameworks -F/nix/store/qfz0c82zhf12x43jxcwhvfrg6f5d4s
g1-apple-framework-CoreBluetooth/Library/Frameworks -F/nix/store/x54lc2llkvp2ak7s3kh0dq3yvlnid8as-apple-framework-IOKit/Library/Frameworks -F/nix/store/gs1c9qkz6b8al4aif7z8wgbl7hxvq1z0-apple-framework-IOSurface/L
ibrary/Frameworks -F/nix/store/5ih2g3461xn7n4dwjzfj4z9kp1va13gi-apple-framework-SystemConfiguration/Library/Frameworks -F/nix/store/9l7ckyy8lrlgbcf7rf93iiikgxv6g3n9-apple-framework-Security/Library/Frameworks -F/
nix/store/qhsgpzs9xsjrrmr2c2gbpgxl9lihkbnw-apple-framework-CoreServices/Library/Frameworks -F/nix/store/j9m3c20vw2sc3z706r34s94whxirwr76-apple-framework-CFNetwork/Library/Frameworks -F/nix/store/hc95skcs2rqf6jyw0
1j075pljg4i060j-apple-framework-CoreAudio/Library/Frameworks -F/nix/store/l66hcpw1cpwrbks2gln8fijq7glmb15z-apple-framework-CoreData/Library/Frameworks -F/nix/store/jvy4anwliaqn66yf8phdc2ai13977mwa-apple-framework
-CoreFoundation/Library/Frameworks -F/nix/store/ii38vbr1rkkmyi0bpfma28n6kawm509z-apple-framework-DiskArbitration/Library/Frameworks -F/nix/store/p05ad4h6lpkm3g7gqfg11zjkr2b13grm-apple-framework-NetFS/Library/Fram
eworks -F/nix/store/zfmqzjnykhdxg5s2ryalixc2avj3l6k7-apple-framework-OpenDirectory/Library/Frameworks -F/nix/store/3cikf6yzkrfj1aar45s9q7mna9008y3i-apple-framework-ServiceManagement/Library/Frameworks -F/nix/stor
e/jplkm2kw2ny8wayp7n7jd3hc3dqyi9xz-apple-framework-CoreText/Library/Frameworks -F/nix/store/b69ikp6xf9ljng8lvylia91rfpcwb22d-apple-framework-ImageIO/Library/Frameworks -F/nix/store/qhhfwlbwaz5ac0xqsvvg0fn61h6rkwq
l-apple-framework-CoreFoundation/Library/Frameworks
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libjava.a(jni_util.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libjava.a(check_version.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libnet.a(net_util.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libnio.a(nio_util.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libzip.a(zip_util.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/svm/clibraries/darwin-amd64/libjvm.a(JvmFuncs.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libjava.a(FileDescriptor_md.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libjava.a(FileInputStream.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libjava.a(FileOutputStream.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libjava.a(UnixFileSystem_md.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libjava.a(Shutdown.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libnet.a(Inet4AddressImpl.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libnet.a(Inet4Address.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libnet.a(Inet6AddressImpl.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libnet.a(Inet6Address.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libnet.a(InetAddress.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libnet.a(NetworkInterface.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libjava.a(TimeZone.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libzip.a(CRC32.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libzip.a(Inflater.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libnio.a(IOUtil.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libnio.a(NativeThread.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libnio.a(UnixFileDispatcherImpl.o)) was built for newer macOS version (11.0) than being linked (10.1
2)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libnio.a(UnixNativeDispatcher.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/svm/clibraries/darwin-amd64/liblibchelper.a(timeZone.o)) was built for newer macOS version (11.0) than being linked (10.
12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/svm/clibraries/darwin-amd64/liblibchelper.a(cpuid.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/svm/clibraries/darwin-amd64/liblibchelper.a(varargs_helper.o)) was built for newer macOS version (11.0) than being linke
d (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/svm/clibraries/darwin-amd64/libdarwin.a(foundation.o)) was built for newer macOS version (11.0) than being linked (10.12
)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libnet.a(net_util_md.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libjava.a(canonicalize_md.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/svm/clibraries/darwin-amd64/libjvm.a(JvmFuncsFallbacks.o)) was built for newer macOS version (11.0) than being linked (1
0.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libjava.a(io_util_md.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libjava.a(TimeZone_md.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libjava.a(jni_util_md.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libjava.a(io_util.o)) was built for newer macOS version (11.0) than being linked (10.12)
ld: warning: object file (/nix/store/ilvxwjpcjmh83kkif1xc3ya1a0qjkx46-graalvm-ce-22.0.0/lib/static/darwin-amd64/libjava.a(path_util.o)) was built for newer macOS version (11.0) than being linked (10.12)
Undefined symbols for architecture x86_64:
  "____chkstk_darwin", referenced from:
      _InflateFully in libzip.a(zip_util.o)
      _checkCPUFeatures in liblibchelper.a(cpuid.o)
      _readBytes in libjava.a(io_util.o)
      _writeBytes in libjava.a(io_util.o)
      _collapse in libjava.a(path_util.o)
ld: symbol(s) not found for architecture x86_64
clang-16: error: linker command failed with exit code 1 (use -v to see invocation)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.image.NativeImageViaCC.handleLinkerFailure(NativeImageViaCC.java:201)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.image.NativeImageViaCC.runLinkerCommand(NativeImageViaCC.java:151)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.image.NativeImageViaCC.write(NativeImageViaCC.java:117)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:732)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:537)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:526)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:701)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.start(NativeImageGeneratorRunner.java:140)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:95)

error: builder for '/nix/store/rj40lzpjd63sc2ammxxvylqs6lfrcc5r-graalvm-ce-22.0.0.drv' failed with exit code 1
```

### Notify maintainers
GraalVM package maintainers (copied from another issue as I couldn't find the list elsewhere, sorry if I have the wrong people):
@bandresen
@volth
@hlolli
@glittershark
@babariviere
@ericdallo
@thiagokokada

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-darwin"`
 - host os: `Darwin 23.4.0, macOS 10.16`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.10.3`
 - channels(djm): `"home-manager, nixpkgs, unstable"`
 - nixpkgs: `/Users/djm/.nix-defexpr/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

So it seems the issue is that `native-image` is now compiled with SDK 11.0 in `x86_64-darwin`. I opened a PR changing it in https://github.com/NixOS/nixpkgs/pull/299203, however it still doesn't work with the following errors:

```
[1/8] Initializing...                                                                                    (0.0s @ 0.10GB)
Error: Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/ybqjzlwc8iazf693ffkvifj23zlsqf8p-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFURL.h:1189:83: error: expected ','
Error: Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/ybqjzlwc8iazf693ffkvifj23zlsqf8p-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFURL.h:1190:98: error: expected ','
Error: Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/ybqjzlwc8iazf693ffkvifj23zlsqf8p-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFURL.h:1200:85: error: expected ','
Error: Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/182hz8xnj1kdp7h3bi3cqj1fs1kdm90r-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSOrderedCollectionChange.h:51:11: error: expected ';' after method prototype
Error: Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/182hz8xnj1kdp7h3bi3cqj1fs1kdm90r-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSBundle.h:156:21: error: expected ';' after method prototype
Error: Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/182hz8xnj1kdp7h3bi3cqj1fs1kdm90r-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSCalendar.h:115:21: error: expected ';' after method prototype
Error: Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/182hz8xnj1kdp7h3bi3cqj1fs1kdm90r-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSUnit.h:96:21: error: expected ';' after method prototype
Error: Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/182hz8xnj1kdp7h3bi3cqj1fs1kdm90r-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSUnit.h:97:20: error: expected ';' after method prototype
Error: Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/182hz8xnj1kdp7h3bi3cqj1fs1kdm90r-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSMeasurement.h:22:21: error: expected ';' after method prototype
Error: Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/182hz8xnj1kdp7h3bi3cqj1fs1kdm90r-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSLocale.h:89:21: error: expected ';' after method prototype
Error: Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/182hz8xnj1kdp7h3bi3cqj1fs1kdm90r-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSURL.h:395:82: error: expected ','
Error: Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/182hz8xnj1kdp7h3bi3cqj1fs1kdm90r-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSURL.h:396:97: error: expected ','
Error: Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/182hz8xnj1kdp7h3bi3cqj1fs1kdm90r-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSURL.h:402:84: error: expected ','
Error: Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/182hz8xnj1kdp7h3bi3cqj1fs1kdm90r-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSProxy.h:28:28: error: expected ';' after method prototype
Error: Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/182hz8xnj1kdp7h3bi3cqj1fs1kdm90r-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSProxy.h:29:28: error: expected ';' after method prototype
Error: Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/i9rkqff9chanrbhqfb314nivcd7c2hz8-apple-framework-Security-11.0.0/Library/Frameworks/Security.framework/Headers/SecBase.h:142:73: error: expected ';' after top level declarator
Error: Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/i9rkqff9chanrbhqfb314nivcd7c2hz8-apple-framework-Security-11.0.0/Library/Frameworks/Security.framework/Headers/SecBase.h:148:77: error: expected ';' after top level declarator
Error: Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/i9rkqff9chanrbhqfb314nivcd7c2hz8-apple-framework-Security-11.0.0/Library/Frameworks/Security.framework/Headers/SecBase.h:154:35: error: expected ';' after top level declarator
Error: Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/i9rkqff9chanrbhqfb314nivcd7c2hz8-apple-framework-Security-11.0.0/Library/Frameworks/Security.framework/Headers/SecBase.h:163:27: error: expected ')'
com.oracle.svm.core.util.UserError$UserException: Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/ybqjzlwc8iazf693ffkvifj23zlsqf8p-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFURL.h:1189:83: error: expected ','
Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/ybqjzlwc8iazf693ffkvifj23zlsqf8p-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFURL.h:1190:98: error: expected ','
Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/ybqjzlwc8iazf693ffkvifj23zlsqf8p-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFURL.h:1200:85: error: expected ','
Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/182hz8xnj1kdp7h3bi3cqj1fs1kdm90r-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSOrderedCollectionChange.h:51:11: error: expected ';' after method prototype
Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/182hz8xnj1kdp7h3bi3cqj1fs1kdm90r-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSBundle.h:156:21: error: expected ';' after method prototype
Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/182hz8xnj1kdp7h3bi3cqj1fs1kdm90r-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSCalendar.h:115:21: error: expected ';' after method prototype
Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/182hz8xnj1kdp7h3bi3cqj1fs1kdm90r-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSUnit.h:96:21: error: expected ';' after method prototype
Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/182hz8xnj1kdp7h3bi3cqj1fs1kdm90r-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSUnit.h:97:20: error: expected ';' after method prototype
Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/182hz8xnj1kdp7h3bi3cqj1fs1kdm90r-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSMeasurement.h:22:21: error: expected ';' after method prototype
Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/182hz8xnj1kdp7h3bi3cqj1fs1kdm90r-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSLocale.h:89:21: error: expected ';' after method prototype
Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/182hz8xnj1kdp7h3bi3cqj1fs1kdm90r-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSURL.h:395:82: error: expected ','
Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/182hz8xnj1kdp7h3bi3cqj1fs1kdm90r-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSURL.h:396:97: error: expected ','
Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/182hz8xnj1kdp7h3bi3cqj1fs1kdm90r-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSURL.h:402:84: error: expected ','
Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/182hz8xnj1kdp7h3bi3cqj1fs1kdm90r-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSProxy.h:28:28: error: expected ';' after method prototype
Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/182hz8xnj1kdp7h3bi3cqj1fs1kdm90r-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSProxy.h:29:28: error: expected ';' after method prototype
Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/i9rkqff9chanrbhqfb314nivcd7c2hz8-apple-framework-Security-11.0.0/Library/Frameworks/Security.framework/Headers/SecBase.h:142:73: error: expected ';' after top level declarator
Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/i9rkqff9chanrbhqfb314nivcd7c2hz8-apple-framework-Security-11.0.0/Library/Frameworks/Security.framework/Headers/SecBase.h:148:77: error: expected ';' after top level declarator
Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/i9rkqff9chanrbhqfb314nivcd7c2hz8-apple-framework-Security-11.0.0/Library/Frameworks/Security.framework/Headers/SecBase.h:154:35: error: expected ';' after top level declarator
Error compiling query code (in /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c). Compiler command '/nix/store/yjzqlbr324ijwya7wgja90f6apjcfq77-clang-wrapper-16.0.6/bin/cc -Wall -Werror -Wno-tautological-compare -ObjC -o /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives /var/folders/ck/_4mg843x0_g3k2ycvg1vlm3800009d/T/SVM-7289511215535823425/PosixDirectives.c' output included error: /nix/store/i9rkqff9chanrbhqfb314nivcd7c2hz8-apple-framework-Security-11.0.0/Library/Frameworks/Security.framework/Headers/SecBase.h:163:27: error: expected ')'
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.util.UserError.abort(UserError.java:138)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.c.NativeLibraries.reportErrors(NativeLibraries.java:380)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.processNativeLibraryImports(NativeImageGenerator.java:1791)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.setupNativeLibraries(NativeImageGenerator.java:1237)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.setupNativeImage(NativeImageGenerator.java:979)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:569)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:537)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:526)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:701)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.start(NativeImageGeneratorRunner.java:140)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:95)
```

I think there is a missing define somewhere, but I am not sure what it is missing.

--- a/pkgs/development/compilers/graalvm/community-edition/buildGraalvm.nix
+++ b/pkgs/development/compilers/graalvm/community-edition/buildGraalvm.nix
@@ -165,33 +165,27 @@ let
       echo "Testing GraalVM"
       $out/bin/java -XX:+UnlockExperimentalVMOptions -XX:+EnableJVMCI -XX:+UseJVMCICompiler HelloWorld | fgrep 'Hello World'
 
-      # Workaround GraalVM issue where the builder does not have access to the
-      # environment variables since 21.0.0
-      # Only needed for native-image tests
-      # https://github.com/oracle/graal/pull/6095
-      # https://github.com/oracle/graal/pull/6095
-      # https://github.com/oracle/graal/issues/7502
-      export NATIVE_IMAGE_DEPRECATED_BUILDER_SANITATION="true";
+      extraNativeImageArgs="$(export -p | sed -n 's/^declare -x \([^=]\+\)=.*$/ -E\1/p' | tr -d \\n)"
 
       echo "Ahead-Of-Time compilation"
-      $out/bin/native-image -H:+UnlockExperimentalVMOptions -H:-CheckToolchain -H:+ReportExceptionStackTraces HelloWorld
+      $out/bin/native-image -H:+UnlockExperimentalVMOptions -H:-CheckToolchain -H:+ReportExceptionStackTraces -march=compatibility $extraNativeImageArgs HelloWorld
       ./helloworld | fgrep 'Hello World'
 
       ${# --static is only available in Linux
       lib.optionalString (stdenv.isLinux && !useMusl) ''
         echo "Ahead-Of-Time compilation with -H:+StaticExecutableWithDynamicLibC"
-        $out/bin/native-image -H:+UnlockExperimentalVMOptions -H:+StaticExecutableWithDynamicLibC HelloWorld
+        $out/bin/native-image -H:+UnlockExperimentalVMOptions -H:+StaticExecutableWithDynamicLibC -march=compatibility $extraNativeImageArgs HelloWorld
         ./helloworld | fgrep 'Hello World'
 
         echo "Ahead-Of-Time compilation with --static"
-        $out/bin/native-image --static HelloWorld
+        $out/bin/native-image $extraNativeImageArgs -march=compatibility --static HelloWorld
         ./helloworld | fgrep 'Hello World'
       ''}
 
       ${# --static is only available in Linux
       lib.optionalString (stdenv.isLinux && useMusl) ''
         echo "Ahead-Of-Time compilation with --static and --libc=musl"
-        $out/bin/native-image --static HelloWorld --libc=musl
+        $out/bin/native-image $extraNativeImageArgs -march=compatibility --libc=musl --static HelloWorld
         ./helloworld | fgrep 'Hello World'
       ''}
 

--- a/pkgs/development/compilers/graalvm/community-edition/buildGraalvm.nix
+++ b/pkgs/development/compilers/graalvm/community-edition/buildGraalvm.nix
@@ -51,7 +51,13 @@ let
     "meta"
   ];
 
-  cLibs = lib.optionals stdenv.isLinux (
+  stdenv' =
+    if stdenv.isDarwin then
+      darwin.apple_sdk_11_0.stdenv
+    else
+      stdenv;
+
+  cLibs = lib.optionals stdenv'.isLinux (
     [ glibc zlib.static ]
     ++ lib.optionals (!useMusl) [ glibc.static ]
     ++ lib.optionals useMusl [ musl ]
@@ -61,14 +67,14 @@ let
   # GraalVM 21.3.0+ expects musl-gcc as <system>-musl-gcc
   musl-gcc = (runCommandCC "musl-gcc" { } ''
     mkdir -p $out/bin
-    ln -s ${lib.getDev musl}/bin/musl-gcc $out/bin/${stdenv.hostPlatform.system}-musl-gcc
+    ln -s ${lib.getDev musl}/bin/musl-gcc $out/bin/${stdenv'.hostPlatform.system}-musl-gcc
   '');
-  binPath = lib.makeBinPath (lib.optionals useMusl [ musl-gcc ] ++ [ stdenv.cc ]);
+  binPath = lib.makeBinPath (lib.optionals useMusl [ musl-gcc ] ++ [ stdenv'.cc ]);
 
   runtimeLibraryPath = lib.makeLibraryPath
     ([ cups ] ++ lib.optionals gtkSupport [ cairo glib gtk3 ]);
 
-  graalvm-ce = stdenv.mkDerivation ({
+  graalvm-ce = stdenv'.mkDerivation ({
     pname = "graalvm-ce";
 
     unpackPhase = ''
@@ -107,7 +113,7 @@ let
       ++ lib.optional stdenv.isLinux autoPatchelfHook;
 
     propagatedBuildInputs = [ setJavaClassPath zlib ]
-      ++ lib.optional stdenv.isDarwin darwin.apple_sdk.frameworks.Foundation;
+      ++ lib.optional stdenv.isDarwin darwin.apple_sdk_11_0.frameworks.Foundation;
 
     buildInputs = lib.optionals stdenv.isLinux [
       alsa-lib # libasound.so wanted by lib/libjsound.so

--- a/pkgs/development/compilers/graalvm/community-edition/buildGraalvm.nix
+++ b/pkgs/development/compilers/graalvm/community-edition/buildGraalvm.nix
@@ -51,13 +51,7 @@ let
     "meta"
   ];
 
-  stdenv' =
-    if stdenv.isDarwin then
-      darwin.apple_sdk_11_0.stdenv
-    else
-      stdenv;
-
-  cLibs = lib.optionals stdenv'.isLinux (
+  cLibs = lib.optionals stdenv.isLinux (
     [ glibc zlib.static ]
     ++ lib.optionals (!useMusl) [ glibc.static ]
     ++ lib.optionals useMusl [ musl ]
@@ -67,14 +61,14 @@ let
   # GraalVM 21.3.0+ expects musl-gcc as <system>-musl-gcc
   musl-gcc = (runCommandCC "musl-gcc" { } ''
     mkdir -p $out/bin
-    ln -s ${lib.getDev musl}/bin/musl-gcc $out/bin/${stdenv'.hostPlatform.system}-musl-gcc
+    ln -s ${lib.getDev musl}/bin/musl-gcc $out/bin/${stdenv.hostPlatform.system}-musl-gcc
   '');
-  binPath = lib.makeBinPath (lib.optionals useMusl [ musl-gcc ] ++ [ stdenv'.cc ]);
+  binPath = lib.makeBinPath (lib.optionals useMusl [ musl-gcc ] ++ [ stdenv.cc ]);
 
   runtimeLibraryPath = lib.makeLibraryPath
     ([ cups ] ++ lib.optionals gtkSupport [ cairo glib gtk3 ]);
 
-  graalvm-ce = stdenv'.mkDerivation ({
+  graalvm-ce = stdenv.mkDerivation ({
     pname = "graalvm-ce";
 
     unpackPhase = ''

--- a/pkgs/development/compilers/graalvm/community-edition/default.nix
+++ b/pkgs/development/compilers/graalvm/community-edition/default.nix
@@ -1,23 +1,28 @@
 { lib
-, stdenv
-, callPackage
-, fetchurl
+, pkgs
 }:
 
+lib.makeScope pkgs.newScope (self:
 {
-  buildGraalvm = callPackage ./buildGraalvm.nix;
+  stdenv =
+    if pkgs.stdenv.isDarwin then
+      pkgs.darwin.apple_sdk_11_0.stdenv
+    else
+      pkgs.stdenv;
 
-  buildGraalvmProduct = callPackage ./buildGraalvmProduct.nix;
+  buildGraalvm = self.callPackage ./buildGraalvm.nix;
 
-  graalvm-ce = callPackage ./graalvm-ce { };
+  buildGraalvmProduct = self.callPackage ./buildGraalvmProduct.nix;
 
-  graalvm-ce-musl = callPackage ./graalvm-ce { useMusl = true; };
+  graalvm-ce = self.callPackage ./graalvm-ce { };
 
-  graaljs = callPackage ./graaljs { };
+  graalvm-ce-musl = self.callPackage ./graalvm-ce { useMusl = true; };
 
-  graalnodejs = callPackage ./graalnodejs { };
+  graaljs = self.callPackage ./graaljs { };
 
-  graalpy = callPackage ./graalpy { };
+  graalnodejs = self.callPackage ./graalnodejs { };
 
-  truffleruby = callPackage ./truffleruby { };
-}
+  graalpy = self.callPackage ./graalpy { };
+
+  truffleruby = self.callPackage ./truffleruby { };
+})

